velox/docs/functions/spark/string.rst

```diff
@@ -179,6 +179,22 @@ Unless specified otherwise, all functions return NULL if at least one of the arg
         SELECT substring('Spark SQL', -10, 3); -- "Sp"
         SELECT substring('Spark SQL', -20, 3); -- ""
 
+.. spark:function:: substring_index(string, delim, count) -> varchar
+
+    Returns the substring from ``string`` before ``count`` occurrences of the delimiter ``delim``.
+    If ``count`` is positive, everything to the left of the final delimiter
+    (counting from the left) is returned. If ``count`` is negative, everything to the right
+    of the final delimiter (counting from the right) is returned. If ``count`` is 0, empty string
+    is returned. ``delim`` is case-sensitive. ::
+
+        SELECT substring_index('Spark.SQL', '.', 1); -- "Spark"
+        SELECT substring_index('Spark.SQL', '.', 0); -- ""
+        SELECT substring_index('Spark.SQL', '.', -1); -- "SQL"
+        SELECT substring_index('TEST.Spark.SQL', '.',2); -- "TEST.Spark"
```

Comment:
Would you also add examples where count is greater then total number of occurrences? 

```
substring_index('TEST.Spark.SQL', '.', 10)

substring_index('TEST.Spark.SQL', '.', -12)
```

Reply:
Added.